.github/workflows/ci.yml

```diff
@@ -48,7 +48,7 @@ jobs:
       matrix:
         java-version: ["8", "11", "17", "21"]
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # 4.1.7
```

Comment:
Why use this specific commit

Reply:
More granular control over version numbers. If not necessary, I will update to @v4 version

Reply:
done!